### PR TITLE
My commit

### PR DESCRIPTION
**Pull request by ValyaKovalenko**

### CODE REVIEW

File: dz-9/src/main/java/org/example/Man.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public Man(String firstName, String lastName, int age) {
+        super(firstName, lastName, age, "male");
+    }
+
+    // Геттер та сеттер для partnerLastName
+    public String getPartnerLastName() {
+        return partnerLastName;
+    }
+
+    public void setPartnerLastName(String partnerLastName) {
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Man.java]: This method redundant here

File: dz-9/src/main/java/org/example/Man.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+public class Man extends Person {
+    private String partnerLastName;
+
+    public Man(String firstName, String lastName, int age) {
+        super(firstName, lastName, age, "male");
+    }
+
+    // Геттер та сеттер для partnerLastName
+    public String getPartnerLastName() {
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Man.java]: This method redundant here

File: dz-9/src/main/java/org/example/Man.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package org.example;
+
+public class Man extends Person {
+    private String partnerLastName;
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Man.java]: This field redundant here

File: dz-9/src/main/java/org/example/Man.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        woman.setLastName(getLastName());
+        setPartnerLastName(woman.getLastName());
+    }
+
+    // Метод для розірвання партнерства (повернення до попереднього прізвища)
+    public void deregisterPartnership(boolean returnToPreviousLastName) {
+        if (returnToPreviousLastName) {
+            Woman partner = new Woman("", partnerLastName, 0);
+            setLastName(partner.getLastName());
+            partner.setLastName(partnerLastName);
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Man.java]: This code do nothing except setting `partnerLastName` for current `Men` but for men we do not need this

File: dz-9/src/main/java/org/example/Person.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package org.example;
+public class Person {
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Person.java]: Should this class be abstract?

File: dz-9/src/main/java/org/example/Person.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
+    // Метод для перевірки, чи особа на пенсії
+    public boolean isRetired() {
+        return age >= pensionAge;
+    }
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Person.java:52]: Add abstract method `deregisterPartnership`

File: dz-9/src/main/java/org/example/Person.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
+package org.example;
+public class Person {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private String gender;
+    private int pensionAge;
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Person.java:7]: Add field `Person partner` and set it in `registerPartnership ` and `deregisterPartnership`

File: dz-9/src/main/java/org/example/Woman.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    public void setPartnerLastName(String partnerLastName) {
+        this.partnerLastName = partnerLastName;
+    }
+
+    // Метод для розірвання партнерства (повернення до попереднього прізвища)
+    public void deregisterPartnership(boolean returnToPreviousLastName) {
+        if (returnToPreviousLastName) {
+            Man partner = new Man("", partnerLastName, 0);
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Woman.java]: The same as for `Men` class this code does almost nothing

File: dz-9/src/main/java/org/example/Woman.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
+package org.example;
+public class Woman extends Person {
+    private String partnerLastName;
```
Comment (yurii-voronenko) [inline dz-9/src/main/java/org/example/Woman.java:3]: May be better to use `initialLastName` for woman and set it in constructor, you can use this field in `deregisterPartnership` to revert lastName

**Review by yurii-voronenko — CHANGES_REQUESTED** [review]

File: dz-9/src/main/java/org/example/Man.java (hunk 1 above)
Comment (ValyaKovalenko) [reply dz-9/src/main/java/org/example/Man.java]: deleted

File: dz-9/src/main/java/org/example/Man.java (hunk 2 above)
Comment (ValyaKovalenko) [reply dz-9/src/main/java/org/example/Man.java]: deleted

File: dz-9/src/main/java/org/example/Man.java (hunk 3 above)
Comment (ValyaKovalenko) [reply dz-9/src/main/java/org/example/Man.java]: The field partherLastName deleted (row 4)

File: dz-9/src/main/java/org/example/Man.java (hunk 4 above)
Comment (ValyaKovalenko) [reply dz-9/src/main/java/org/example/Man.java]: ok, deleted for `Man`

File: dz-9/src/main/java/org/example/Woman.java (hunk 8 above)
Comment (ValyaKovalenko) [reply dz-9/src/main/java/org/example/Woman.java]: ok, changed

**Pull request merged**